### ISSUE TITLE
Azure AD : Too many redirects because of unsuccessful authentication.

### ISSUE THREAD

**Issue by arun-898** [vouch/vouch-proxy#495]
I am trying to set up vouch  proxy with Azure AD B2C with Drupal app using the following [example ](https://github.com/vouch/vouch-proxy/blob/master/examples/nginx/single-file/nginx_with_vouch_ssl.conf)

I have tried the following two providers.
**Azure provider:**
This fails immediately at `/auth` request because of the missing access token, so I decided to use ADFS
Error: `/auth Error while retrieving user info after successful login at the OAuth provider: oauth2: server response missing access_token`

**ADFS Provider:** 
it results in too many redirects, because of failed authentication it keeps on redirecting to vouch proxy login URL. (all request redirects to 302) All the logs for ADFS are attached [here](https://gist.github.com/arun-898/a060669bff02ff589664d51787885a19)

Can anyone please help me to resolve this issue?

**Comment by bnfinet** [comment]
@arun-898 thanks for the detailed report.

It may be related to #445, that appears to be the same behavior.

I'm not an ADFS or Azure user and I'm not in a good position to assess this bug.

Are you able to populate the `Username` field as per the following?  I'm wondering if that would change the behavior.

`2022-08-16T13:14:52.481Z	DEBUG	User Obj: &{Username: Name: Email:test_user.test@gmail.com CreatedOn:0 LastUpdate:0 ID:0 TeamMemberships:[]}`

In general I'd like to migrate VP to use to `sub` for it's primary user identifier but that needs some thought and work, which is partly why #445 and other related PRs have sat there for so long.

**Comment by arun-898** [comment]
Thanks @bnfinet  for the quick response, I am trying to populate Username with available attributes.

**Comment by arun-898** [comment]
I was not able to populate the username but I noticed some issues with claims.
After removing claims from the vouch config and Nginx config  I was able to navigate to my app URL with `testing: true`.
but when I keep `testing: false` it results in too many redirects

**Comment by bnfinet** [comment]
@arun-898 I notice you're using
v0.27.1 of Vouch Proxy.  Could you please update your version of VP 

If you still have problems after upgrade, please do offer logs and config.

**Comment by arun-898** [comment]
Thanks @bnfinet for the info, I have updated the latest VP tag `v0.37.3`, I will share the latest logs.

**Comment by arun-898** [comment]
@bnfinet In the latest logs also I am getting the same error.
`2022-08-17T19:00:06.452Z	WARN	no User found in jwt`
is it possible to override other files in the vouch proxy repository, I am able to override  config.yml only
`  image: quay.io/vouch/vouch-proxy:alpine-latest
    volumes:
      - ./vouch/config/config.yml:/config/config.yml`
      
Can you please suggest how I can override the required files, I can then debug further for populating the username.
id_token returned from Azure does not have UPN field, so most likely I would have to alter
vouch/pkg/structs/structs.go // getname from Name field instead of UPN field.
`func (u *ADFSUser) PrepareUserData() {
	u.Username = u.Name
}`

**Comment by bnfinet** [comment]
@arun-898 could you please publish those logs and the current config you're using to a gist as before?

> Can you please suggest how I can override the required files

The best way to do that is to download the vouch-proxy source code from this git repo and then run `/do.sh dbuildalpine` which will build a new alpine docker image which you can use.

**Comment by arun-898** [comment]
@bnfinet I have updated the logs and config for Vouch Proxy v0.37.3 in the ticket description, please have a look at the logs.

**Comment by bnfinet** [comment]
@arun-898 I'm glad it's working for you with those changes.  I think that confirms what we already know from #445.

I'd be very curious to understand why it was working with `testing: true` but not with `testing: false`.

**Comment by arun-898** [comment]
@bnfinet yes Changes added here #445 will be required for ADFS user object as well until we move to `sub` logic  #310 

> I'd be very curious to understand why it was working with testing: true but not with testing: false.
Please ignore this, I misunderstood this with `testing: false` I was able to redirect to app URL but when I click on the app URL, it was again redirecting me to the IDP. This is fixed now with latest config.
